numba_dpex/core/passes/parfor_lowering_pass.py

```diff
@@ -42,24 +50,16 @@ class ParforLowerImpl:
     for a parfor and submits it to a queue.
     """
 
-    def _submit_parfor_kernel(
-        self,
-        lowerer,
-        kernel_fn,
-        loop_ranges,
-    ):
-        """
-        Adds the call to the dpex kernel function from the main function.
-        """
+    def _prepare_queue_num_args(self, kernel_fn, lowerer):
         keep_alive_kernels.append(kernel_fn.kernel)
 
         # Helper class that generates the LLVM IR values inside the current LLVM
         # module that are needed to submit the kernel to a queue.
-        ir_builder = KernelLaunchIRBuilder(lowerer, kernel_fn.kernel)
+        self.ir_builder = KernelLaunchIRBuilder(lowerer, kernel_fn.kernel)
```

Comment:
rename to `kernel_builder` to be clear that it is not directly a llvmlite.ir..IRBuilder